pkg/skaffold/build/acr/acr_build.go

```diff
@@ -0,0 +1,190 @@
+	}
+
+	err = blob.UploadFileToBlob()
+	if err != nil {
+		return "", errors.Wrap(err, "upload file to blob")
+	}
+
+	//acr needs the image tag formatted as <repository>:<tag>
+	imageTag = getImageTagWithoutFQDN(imageTag)
+
+	buildRequest := cr.DockerBuildRequest{
+		ImageNames:     &[]string{imageTag},
+		IsPushEnabled:  util.BoolPtr(true),
+		SourceLocation: result.RelativePath,
+		Platform: &cr.PlatformProperties{
+			Variant:      cr.V8,
+			Os:           cr.Linux,
+			Architecture: cr.Amd64,
+		},
+		DockerFilePath: &artifact.DockerArtifact.DockerfilePath,
+		Type:           cr.TypeDockerBuildRequest,
+	}
+	future, err := client.ScheduleRun(ctx, resourceGroup, registryName, buildRequest)
+	if err != nil {
+		return "", errors.Wrap(err, "schedule build request")
+	}
+
+	run, err := future.Result(*client)
+	if err != nil {
+		return "", errors.Wrap(err, "get run id")
+	}
+	runID := *run.RunID
+
+	runsClient := cr.NewRunsClient(b.SubscriptionID)
+	runsClient.Authorizer = client.Authorizer
+	logURL, err := runsClient.GetLogSasURL(ctx, resourceGroup, registryName, runID)
+	if err != nil {
+		return "", errors.Wrap(err, "get log url")
+	}
+
+	err = streamBuildLogs(*logURL.LogLink, out)
+	if err != nil {
+		return "", errors.Wrap(err, "polling build status")
+	}
+
+	return imageTag, nil
+}
+
+func streamBuildLogs(logURL string, out io.Writer) error {
+	offset := int32(0)
+	for {
+		resp, err := http.Get(logURL)
+		if err != nil {
+			return err
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			//if blob is not available yet, try again
+			time.Sleep(2 * time.Second)
+			continue
+		}
+
+		scanner := bufio.NewScanner(resp.Body)
+		line := int32(0)
+		for scanner.Scan() {
+			if line >= offset {
+				out.Write(scanner.Bytes())
+				out.Write([]byte("\n"))
+				offset++
+			}
+			line++
+		}
+		resp.Body.Close()
+
+		if offset > 0 {
+			switch resp.Header.Get(BuildStatusHeader) {
+			case "":
+				continue
+			case "internalerror":
+			case "failed":
+				return errors.New("run failed")
+			case "timedout":
+				return errors.New("run timed out")
+			case "canceled":
+				return errors.New("run was canceled")
+			default:
+				return nil
+			}
+		}
+
+		time.Sleep(2 * time.Second)
+	}
+}
+
+func getResourceGroup(ctx context.Context, client *cr.RegistriesClient, registryName string) (string, error) {
+	registryList, err := client.List(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	for _, registry := range registryList.Values() {
+		if strings.ToLower(*registry.Name) == registryName {
+			//registry.ID returns the exact path to the container registry
+			//e.g. /subscriptions/<subscriptionId>/resourceGroups/<resourceGroup>/...
+			//so the resourceGroup is the fourth element of the split
+			return strings.Split(*registry.ID, "/")[4], nil
+		}
+	}
+
+	return "", errors.New("Couldn't find resource group of registry")
+}
+
+func getImageTagWithoutFQDN(imageTag string) string {
+	return imageTag[strings.Index(imageTag, "/")+1:]
```

Comment:
I think theres a better way to do this using the go-containerregistry. Can you look at an example of the other builders

Reply:
Maybe add a test or two?

Reply:
Thanks for the tip with go-containerregistry 👍